src/thor/triplegbuilder.cc

```diff
@@ -1406,7 +1489,10 @@ void TripLegBuilder::Build(
     // we could cache the timezone and just add seconds when the timezone doesnt change
     uint32_t second_of_week = kInvalidSecondsOfWeek;
     if (origin_epoch != 0) {
-      second_of_week = DateTime::second_of_week(origin_epoch + static_cast<uint32_t>(elapsedtime),
+      auto elapsed_seconds = static_cast<uint32_t>(
+          trip_path.node_size() == 0 ? 0
+                                     : trip_path.node().rbegin()->cost().elapsed_cost().seconds());
```

Comment:
we already store elapsed seconds on the nodes of the path as we add them. so we just get the information from there.